AK/Error.h

```diff
@@ -24,7 +25,12 @@ class Error {
 public:
     static Error from_errno(int code) { return Error(code); }
     static Error from_syscall(StringView syscall_name, int rc) { return Error(syscall_name, rc); }
-    static Error from_string_literal(StringView string_literal) { return Error(string_literal); }
+    static Error from_string_view(StringView string_literal) { return Error(string_literal); }
+
+    ALWAYS_INLINE static Error from_string_literal(char const* string_literal)
```

Comment:
why a char* API?

Reply:
Was going to comment on that, same remark. If you're keeping both APIs to change less of the existing code, then it'd be nice to undo a bunch of your `from_string_literal("thing"sv)` => `from_string_literal("")` changes and do something like `from_string_literal("thing"sv)` => `from_string_view(""sv)`. If there's some other reason that makes it so "from_string_literal" should be preferred, leaving a comment on why the new API should not be used (or when it should/shouldn't be used) would be good.
In both cases, it'd be very nice (though not mandatory) to split that specific change + its impact into its own commit.

Reply:
This is used very commonly and having it be `sv`-prefixed everytime isn't really worth it (it says `string_literal` in its name after all). The existing cases should be handled at compile-time with the `__builtin_strlen`.

Reply:
> If there's some other reason that makes it so "from_string_literal" should be preferred, leaving a comment on why the new API should not be used (or when it should/shouldn't be used) would be good.
Will add the comment. The idea is that the `from_string_literal` function will never be used with a string that's dynamic, and therefore doesn't need the same protection as with the other APIs.

Reply:
Shouldn't this use the template magic you're adding in other places instead of a `char const*`?
```suggestion
    template<size_t N>
    ALWAYS_INLINE static Error from_string_literal(char const (&string_literal)[N])
```

Reply:
That's a good idea that doesn't break the existing usecase, will add.